server/protocols/jmap-draft/src/main/java/org/apache/james/jmap/draft/methods/SetMailboxesUpdateProcessor.java

```diff
@@ -175,7 +174,7 @@ private void assertSystemMailboxesAreNotUpdated(Mailbox mailbox, MailboxUpdateRe
     @VisibleForTesting
     <T> boolean requestChanged(Optional<T> requestValue, Optional<T> storeValue) {
         return requestValue
```

Comment:
I still don't see the point of doing a filter without using the value.

the entire method body could be : 

```
return !requestValue.equals(storeValue)
```